pkg/yurtctl/cmd/convert/edgenode.go

```diff
@@ -0,0 +1,399 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package convert
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	nodeutil "k8s.io/kubernetes/pkg/controller/util/node"
+
+	"github.com/alibaba/openyurt/pkg/projectinfo"
+	enutil "github.com/alibaba/openyurt/pkg/yurtctl/util/edgenode"
+	kubeutil "github.com/alibaba/openyurt/pkg/yurtctl/util/kubernetes"
+	strutil "github.com/alibaba/openyurt/pkg/yurtctl/util/strings"
+	"github.com/alibaba/openyurt/pkg/yurthub/healthchecker"
+)
+
+const (
+	kubeletConfigRegularExpression = "\\-\\-kubeconfig=.*kubelet.conf"
+	apiserverAddrRegularExpression = "server: (http(s)?:\\/\\/)?[\\w][\\w]{0,62}(\\.[\\w][-\\w]{0,62})+(:[\\d]{1,5})?"
+	hubHealthzCheckFrequency       = 10 * time.Second
+	failedRetry                    = 5
+	filemode                       = 0666
+	dirmode                        = 0755
+)
+
+// ConvertEdgeNodeOptions has the information required by sub command convert edgenode
+type ConvertEdgeNodeOptions struct {
+	clientSet          *kubernetes.Clientset
+	EdgeNodes          []string
+	YurthubImage       string
+	YurctlServantImage string
+	PodMainfestPath    string
+	JoinToken          string
+	kubeletSvcPath     string
+	openyurtDir        string
+}
+
+// NewConvertEdgeNodeOptions creates a new ConvertEdgeNodeOptions
+func NewConvertEdgeNodeOptions() *ConvertEdgeNodeOptions {
+	return &ConvertEdgeNodeOptions{}
+}
+
+// NewConvertEdgeNodeCmd generates a new sub command convert edgenode
+func NewConvertEdgeNodeCmd() *cobra.Command {
+	c := NewConvertEdgeNodeOptions()
+	cmd := &cobra.Command{
+		Use:   "edgenode",
+		Short: "Converts the kubernetes node to a yurt node",
+		Run: func(cmd *cobra.Command, _ []string) {
+			if err := c.Complete(cmd.Flags()); err != nil {
+				klog.Fatalf("fail to complete the convert edgenode option: %s", err)
+			}
+			if err := c.RunConvertEdgeNode(); err != nil {
+				klog.Fatalf("fail to covert the kubernetes node to a yurt node: %s", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringP("edge-nodes", "e", "",
+		"The list of edge nodes wanted to be convert.(e.g. -e edgenode1,edgenode2)")
+	cmd.Flags().String("yurthub-image", "openyurt/yurthub:latest",
+		"The yurthub image.")
+	cmd.Flags().String("yurtctl-servant-image", "openyurt/yurtctl-servant:latest",
+		"The yurtctl-servant image.")
+	cmd.Flags().String("pod-manifest-path", "",
+		"Path to the directory on edge node containing static pod files.")
+	cmd.Flags().String("join-token", "", "The token used by yurthub for joining the cluster.")
+
```

Comment:
Does it need to add `kubeadm-conf-path` flag here for kubelet service config path?

Reply:
Oh, yes.